src/test/java/net/snowflake/client/jdbc/SnowflakeDriverLatestIT.java

```diff
@@ -944,4 +944,109 @@ private void copyContentFrom(File file1, File file2) throws Exception {
     fOut.close();
     fIn.close();
   }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testPutS3RegionalUrl() throws Throwable {
+    Connection connection = null;
+    File destFolder = tmpFolder.newFolder();
+    String destFolderCanonicalPath = destFolder.getCanonicalPath();
+
+    List<String> supportedAaccounts = Arrays.asList("s3testaccount", "azureaccount");
+    for (String accountName : supportedAaccounts) {
+      try {
+        connection = getConnection(accountName);
+        Statement statement = connection.createStatement();
+
+        // create a stage to put the file in
+        statement.execute("CREATE OR REPLACE STAGE " + testStageName);
```

Comment:
The new connection parameter `ENABLE_STAGE_S3_PRIVATELINK_FOR_US_EAST_1` has not been set to True in this test case. 
```suggestion
        statement.execute("CREATE OR REPLACE STAGE " + testStageName);
        statement.execute("alter session set ENABLE_STAGE_S3_PRIVATELINK_FOR_US_EAST_1=true");
        // assert parameter is set properly
        assertTrue(connection.unwrap(SnowflakeConnectionV1.class).getSFBaseSession().getUseS3RegionalUrl());
```

Reply:
@sfc-gh-mknister Line 985 and 1000 sets the value of the parameter to false and true and tests it.